internal/common/rules/rulematrix.go

```diff
@@ -4,10 +4,27 @@ import (
 	"gonum.org/v1/gonum/mat"
 )
 
+// LinkTypeOption gives an enumerated type for the various link types available for rules
+type LinkTypeOption int
+
+const (
```

Comment:
An explanation of what these mean would be nice

Reply:
Sure added in latest commit